### PR TITLE
Refactor settings modal

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

**Comment by virtuacoplenny** [comment]
Make sure to grab my test changes too. https://github.com/jitsi/jitsi-meet-torture/pull/184.

**Comment by virtuacoplenny** [comment]
The styling seems off to me. I don't see the "settings-pane" class getting set when looking at profile or other settings.

File: react/features/settings/functions.js [hunk 1]
```diff
@@ -60,3 +62,50 @@ export function shouldShowOnlyDeviceSelection() {
...
+    const language = i18next.language || DEFAULT_LANGUAGE;
+    const conference = state['features/base/conference'];
+    const configuredTabs = interfaceConfig.SETTINGS_SECTIONS || [];
+
+    return {
+        currentLanguage: language,
+        followMeEnabled: Boolean(conference.followMeEnabled),
+        languages: LANGUAGES,
+        showLanguageSettings: configuredTabs.includes('language'),
+        showModeratorSettings: configuredTabs.includes('moderator'),
```
Comment (virtuacoplenny) [inline react/features/settings/functions.js]: This also should check for if the user is a moderator. That's why the tests are failing, I think.

Comment (hristoterezov) [reply react/features/settings/functions.js]: Fixed!

**Comment by hristoterezov** [comment]
> The styling seems off to me. I don't see the "settings-pane" class getting set when looking at profile or other settings.

Good catch! It should be fixed now!

**Comment by virtuacoplenny** [comment]
Test failures look legit to me. Maybe because mobile is somehow loading AtlaskitThemeProvider, which I would assume mobile can't handle because it's made for web.

**Pull request merged**